### PR TITLE
[mux] VS test for neigh, route and fdb

### PR DESCRIPTION
**Pull request by prsunny**
**What I did**

VS test to cover:

1. IPv4 and V6 neighbor add and simulate mux state change.  Verify: _Active mux shall have host entry, Standby mux shall have tunnel route_
2. Route add and simulate mux state change. Verify: _Route should have next hop programmed based on mux state (Active shall have neighbor NH, standby shall have tunnel NH)_
3.  ECMP route add and simulate mux state change. Verify: _ECMP group shall be updated based on mux state change, active mux shall have regular nexthop member, standby shall have tunnel nexthop member_
3. Neighbor handing based on FDB learnt. Verify: _Neighbor with FDB learnt on active mux shall have host entry whereas on standby mux shall have tunnel route_

**Why I did it**
unit-test

**How I verified it**

Run `pytest test_mux.py --dvsname=vs --forcedvs`

**Details if related**

### CODE REVIEW

File: tests/test_mux.py [hunk 1]
```diff
@@ -40,45 +46,307 @@ class TestMuxTunnelBase(object):
...
+            fvs = asicdb.get_entry(self.ASIC_NEXTHOP_TABLE, key)
+            if fvs.get("SAI_NEXT_HOP_ATTR_TYPE") == "SAI_NEXT_HOP_TYPE_TUNNEL_ENCAP":
+                tunnel_nh_id = key
+
+        assert tunnel_nh_id
+
+
+    def check_nexthop_in_asic_db(self, asicdb, key, standby=False):
+
+        global tunnel_nh_id
```
Comment (daall) [inline tests/test_mux.py:118]: Is this needed if we're not setting `tunnel_nh_id`?

File: tests/test_mux.py [hunk 2]
```diff
@@ -40,45 +46,307 @@ class TestMuxTunnelBase(object):
...
+
+        fvs = asicdb.get_entry("ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY", key)
+
+        nhg_id = fvs["SAI_ROUTE_ENTRY_ATTR_NEXT_HOP_ID"]
+
+        fvs = asicdb.get_entry("ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP_GROUP", nhg_id)
+
+        assert bool(fvs)
```
Comment (daall) [inline tests/test_mux.py:141]: If you use `wait_for_entry` these two lines can be combined

File: tests/test_mux.py [hunk 3]
```diff
@@ -40,45 +46,307 @@ class TestMuxTunnelBase(object):
...
+
+        keys = asicdb.get_keys("ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP_GROUP_MEMBER")
+
+        # Two NH group members are expected to be added
+        assert len(keys) == 2
```
Comment (daall) [inline tests/test_mux.py:146]: Same with these two lines and `wait_for_n_keys`

Comment (prsunny) [reply tests/test_mux.py]: done

File: tests/test_mux.py (hunk 1 above)
Comment (prsunny) [reply tests/test_mux.py:118]: this is need as its used for checking in this function. Only one tunnel_nh_id exists in the system, so using it as a global var.

Comment (daall) [reply tests/test_mux.py:118]: That makes sense, but is `global` still needed if we're just reading the variable?

Comment (prsunny) [reply tests/test_mux.py]: I see your point. Addressed.

**Review by daall — APPROVED** [review]

**Pull request merged**